src/lib.rs

```diff
@@ -77,14 +77,37 @@ mod rustfluent {
         ) -> PyResult<String> {
             self.bundle.set_use_isolating(use_isolating);
 
-            let msg = self
-                .bundle
-                .get_message(identifier)
-                .ok_or_else(|| (PyValueError::new_err(format!("{identifier} not found"))))?;
+            let pattern = if identifier.contains('.') {
+                let parts: Vec<&str> = identifier.splitn(2, '.').collect();
+                let message_id = parts[0];
+                let attribute_name = parts[1];
+
+                let msg = self
+                    .bundle
+                    .get_message(message_id)
+                    .ok_or_else(|| PyValueError::new_err(format!("{message_id} not found")))?;
+
+                let attribute = msg.get_attribute(attribute_name).ok_or_else(|| {
+                    PyValueError::new_err(format!(
+                        "{identifier} - Attribute '{attribute_name}' not found on message '{message_id}'."
+                    ))
+                })?;
 
-            let pattern = msg.value().ok_or_else(|| {
-                PyValueError::new_err(format!("{identifier} - Message has no value.",))
-            })?;
+                // Get the pattern from the attribute
+                // Note: attribute.value() returns &Pattern directly (not Option)
+                // because attributes always have values, unlike messages which can
+                // exist with only attributes (no main value)
+                attribute.value()
+            } else {
+                let msg = self
+                    .bundle
+                    .get_message(identifier)
+                    .ok_or_else(|| PyValueError::new_err(format!("{identifier} not found")))?;
```

Comment:
With the helper method suggested above:

```suggestion
                let msg = self.get_message(identifier)?;
```